cogs/gru_nose.py

```diff
@@ -1,38 +1,55 @@
-import asyncio
 import datetime
 import logging
 import os
 
 import discord
-from discord.ext import tasks, commands
+from discord.ext import tasks
 
-from discord_bot import BASE_PATH
+from cogs import BASE_PATH
+from cogs.base import BaseCog
 
-class GruNosePoster(commands.Cog):
-    channel_id = 594707074609577994
 
-    def __init__(self, bot):
-        self.bot = bot
-        self.gru_nose_poster.start()
+class GruNosePoster(BaseCog):
+
+    def __init__(self, bot) -> None:
+        self.channel_id = os.getenv('GRU_NOSE_CHANNEL')
+        self.channel = None
+        if self.channel_id:
+            self.gru_nose_poster.start()
+        else:
+            logging.warning('No channel id was provided for the Gru Nose Poster')
+        super().__init__(bot)
 
     def cog_unload(self):
         self.gru_nose_poster.cancel()
 
-    @tasks.loop(hours=1.0)
+    @tasks.loop(seconds=5)
```

Comment:
I set this to loop every 5 seconds because after the first check, it will adjust its the next loop time to be the offset from now until 12pm. It technically only loops in 5 seconds the first time the bot is booted. I am still long term testing this.